activemodel/lib/active_model/type/date.rb

```diff
@@ -8,7 +8,7 @@ def type
       end
 
       def type_cast_for_schema(value)
-        "'#{value.to_s(:db)}'"
+        super(value.to_s(:db))
```

Comment:
I'd prefer to change this to `%("#{value.to_s(:db)}")`. The current implementation relies on the exact behavior of the parent method.